tmt/utils.py

```diff
@@ -6183,3 +6190,33 @@ def __exit__(self, *args: Any) -> None:
     @property
     def duration(self) -> datetime.timedelta:
         return self.end_time - self.start_time
+
+
+def retry(
+        func: Callable[..., T],
+        retries: int,
+        delay: int,
+        label: str,
+        logger: tmt.log.Logger,
+        *args: Optional[Any],
+        **kwargs: Optional[Any]
+        ) -> T:
+    """ Retry functionality to be used elsewhere in the code.
+
+    :param func: function to be called with any amount
+    of arguments of Any type, returning the value of type TypeVar
+    :param retries: number of tries to call the function
+    :param delay: amount of seconds to wait before a new try
+    :param label: action to retry
+    """
+    exc = None
+    for i in range(retries):
+        try:
+            return func(*args, **kwargs)
+        except Exception as e:
+            exc = e
+            logger.debug(
+                'retry',
+                f"{label} failed, {retries - i} retries left, trying again in {delay:.2f} seconds")
```

Comment:
Right, yes, we should know about errors, but this way, only the very last one would be recorded. I did not realize the situation is not similar to what `wait()` does, because `wait()` is not interrupted multiple times, trying again...

So, we probably need to log the failures - we can't raise an exception, so we need to log at least. I'd keep the `debug` as it is, it aligns with `wait()` and reports important stuff, but to let the user know, we should add `logger.fail(str(exc))` at least.

Reply:
Nope, it was a placeholder, nothing more.